src/components/views/elements/MemberEventListSummary.js

```diff
@@ -0,0 +1,219 @@
+    },
+
+    getEventSenderName: function(ev) {
+        if (!ev) {
+            return 'undefined';
+        }
+        return ev.sender.name || ev.event.content.displayname || ev.getSender();
+    },
+
+    renderNameList: function(events) {
+        if (events.length === 0) {
+            return null;
+        }
+        let originalNumber = events.length;
+        events = events.slice(0, this.props.summaryLength);
+        let lastEvent = events.pop();
+
+        let names = events.map((ev) => {
+            return this.getEventSenderName(ev);
+        }).join(', ');
+
+        if (names.length === 0) {
+            return this.getEventSenderName(lastEvent);
+        }
+
+        // Special case the last name. ' and ' might be included later
+        // So you have two cases:
+        if (originalNumber <= this.props.summaryLength) {
+            // name1, name2 and name3
+            names += ' and ';
+        } else {
+            // name1, name2, name3 [and 100 others]
+            names += ', ';
+        }
+        return names + this.getEventSenderName(lastEvent);
+    },
+
+    renderSummary: function(joinEvents, leaveEvents) {
+        let joiners = this.renderNameList(joinEvents);
+        let remainingJoiners = joinEvents.length - this.props.summaryLength;
+        let leavers = this.renderNameList(leaveEvents);
+        let remainingLeavers = leaveEvents.length - this.props.summaryLength;
+        let joinSummary = null;
+        if (joiners) {
+            joinSummary = (
+                <span>
+                    {joiners} {remainingJoiners > 0 ? 'and ' + remainingJoiners + ' others ':''}joined the room
+                </span>
+            );
+        }
+        let leaveSummary = '';
+        if (leavers) {
+            leaveSummary = (
+                <span>
+                    {leavers} {remainingLeavers > 0 ? 'and ' + remainingLeavers + ' others ':''}left the room
+                </span>
+            );
+        }
+        return (
+            <span>
+                {joinSummary}{joinSummary && leaveSummary?'; ':''}
+                {leaveSummary}
+            </span>
+        );
+    },
+
+    renderAvatars: function(events) {
+        let avatars = events.slice(0, this.props.avatarsMaxLength).map((e) => {
+            let onClickAvatar = () => {
+                dispatcher.dispatch({
+                    action: 'view_user',
+                    member: e.sender,
+                });
+            };
+            return (
+                <MemberAvatar
+                    key={e.getId()}
+                    member={e.sender}
+                    width={14}
+                    height={14}
+                    onClick={onClickAvatar}
+                />
+            );
+        });
+
+        return (
+            <span>
+                {avatars}
+            </span>
+        );
+    },
+
+    render: function() {
+        let summary = null;
+
+        // Reorder events so that joins come before leaves
```

Comment:
oh, I think I misread before: I thought that the list being passed into `renderEvents` depended on the filtering calculations. Looks like it doesn't, so this is fine.